ecommerce/views/v0/__init__.py

```diff
@@ -413,7 +414,13 @@ def post(self, request, *args, **kwargs):
                 )
             )
             if basket:
-                basket.delete()
+                if basket.has_user_blocked_products(order.purchaser):
```

Comment:
I don't think this check should only be done here. (I know the issue indicated to put this check here specifically - not sure if this was really the intent, though.) Review is a state that most transactions won't hit, so learners will generally not hit the check. Following this pattern, I think it should also run this check for the Accepted state.

That said, this is pretty late in the payment process - at this point we've taken their money (potentially - if it's in Review, we haven't, but we're like 50% of the way there). Should this check be run at an earlier stage of the transaction? If we want to stick with it revolving around the handoff to CyberSource, it might be better to have it run before we send them off to pay. Or, I can see a good argument for having the check way further up at the point where they're adding the course to the cart - if they're in a blocked country, do they even need to be able to try to pay into a certificate? I get an error when attempting to enroll in the free course using my blocked country user, so that might be the better flow.

Reply:
yes, That make much more sense.